packages/dbgpt-serve/src/dbgpt_serve/finance/api/endpoints.py

```
@@ -0,0 +1,150 @@
+    return _build_response(result)
+
+
+@router.post(
+    "/analyze/upload",
+    response_model=AnalyzeResponse,
+    dependencies=[Depends(check_api_key)],
+)
+async def analyze_with_files(
+    company: str = Form(..., description="Company name to analyze."),
+    query: Optional[str] = Form(None),
+    files: List[UploadFile] = File(default_factory=list),
```

Ruff (0.16.1)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)